src/index.ts

```diff
@@ -208,7 +209,7 @@ function startApp(appSchema, path: string) {
   })
 
   app.use("/batch", bodyParser.json(), graphqlBatchHTTPWrapper(graphqlServer))
-  app.use(graphqlServer)
+  app.use(graphqlUploadMiddleware, graphqlServer)
```

Comment:
**suggestion**: I still think we should hoist up the package middleware here, instead of defining a custom middleware, and only call `app.use` if the feature flag is enabled. Makes the change more explicit and ensures that we're not modifying the middleware chain unless that flag is enabled:

```typescript
app.use("/batch", bodyParser.json(), graphqlBatchHTTPWrapper(graphqlServer))

if (ENABLE_GRAPHQL_UPLOAD) {
  app.use(graphqlUploadExpress({
    maxFileSize: GRAPHQL_UPLOAD_MAX_FILE_SIZE_IN_BYTES,
    maxFiles: GRAPHQL_UPLOAD_MAX_FILES,
  }))
}

app.use(graphqlServer)
```

The `graphqlUploadExpress` should call next() unless a request has a `multipart/form-data` content type. I think the default error response of 404 if a client does execute a request with this content-type and the feature flag is disabled is totally fine for this iteration.

What do you think?

Reply:
> I think the default error response of 404 if a client does execute a request with this content-type and the feature flag is disabled is totally fine for this iteration

small clarification: in this scenario, we will not get a 404 error (see screenshot), since `express-graphql` is next in the middleware chain

<img width="944" alt="image" src="https://user-images.githubusercontent.com/3513494/165302103-df3b6c58-e581-422a-b5bb-7cc94edb8654.png">

Reply:
Gotcha. Thanks for the clarification! 👍🏼

I think that's a fine error message for now.